drivers/dma/dma_mcux_edma.c

```diff
@@ -224,6 +229,57 @@ static int dma_mcux_edma_configure(const struct device *dev, uint32_t channel,
 		return -EINVAL;
 	}
 
+	if (!data_size_valid(config->source_data_size)) {
+		LOG_ERR("Source unit size error, %d", config->source_data_size);
+		return -EINVAL;
+	}
+
+	if (!data_size_valid(config->dest_data_size)) {
+		LOG_ERR("Dest unit size error, %d", config->dest_data_size);
+		return -EINVAL;
+	}
+
+	if (block_config->source_gather_en || block_config->dest_scatter_en) {
+		if (config->block_count > CONFIG_DMA_TCD_QUEUE_SIZE) {
+			LOG_ERR("please config DMA_TCD_QUEUE_SIZE as %d", config->block_count);
+			return -EINVAL;
+		}
+	}
+
+#if defined(CONFIG_HAS_MCUX_CACHE)
+	/* Ensure that tranfers whose source is a memory location are located in non-cacheable memory */
+	for (const struct dma_block_config *block_to_check = block_config;
+	     block_to_check != NULL;
+	     block_to_check = block_to_check->next_block) {
+		const size_t buf_len = block_to_check->block_size;
+
+		if (transfer_type == kEDMA_MemoryToMemory || transfer_type == kEDMA_MemoryToPeripheral) {
+			const uint8_t *src_buf = (const uint8_t *)block_to_check->source_address;
+			if (!buffer_in_noncacheable_region(src_buf, buf_len)) {
+				LOG_ERR("DMA transfer source buffer is located in cacheable memory.");
+				return -EINVAL;
+			}
+		}
+		if (transfer_type == kEDMA_MemoryToMemory || transfer_type == kEDMA_PeripheralToMemory) {
+			const uint8_t *dst_buf = (const uint8_t *)block_to_check->dest_address;
+			if (!buffer_in_noncacheable_region(dst_buf, buf_len)) {
+				LOG_ERR("DMA transfer destination buffer is located in cacheable memory.");
+				return -EINVAL;
+			}
+		}
+	}
+#endif // CONFIG_HAS_MCUX_CACHE
```

Comment:
Per [our discussion](https://github.com/zephyrproject-rtos/zephyr/pull/42750#issuecomment-1061102220), let's remove the requirement that buffers are in non-cacheable memory.